### PR TITLE
test(tagging): add coverage for heading-level clamping, SemanticEntry, and classify_element edge cases

### PR DESCRIPTION
**Pull request by mitsuru**
## 対象モジュール

`crates/fulgur/src/tagging.rs` — Tagged PDF セマンティックレイヤー

カバレッジ計測時点（2026-07-20）の最低カバレッジ上位 3 ファイル:

| ファイル | Region カバレッジ |
|----------|-----------------|
| `convert/list_item.rs` | 91.36% |
| `render.rs` | 92.46% |
| `tagging.rs` | **93.72%** ← 本 PR |

`render.rs`・`convert/list_item.rs` は Blitz/Krilla 依存が深く本ルーティン（テスト専用）での unit test 追加が困難なため、純粋ロジックのみで構成される `tagging.rs` を選択しました。

## カバレッジ変化

| 指標 | 変更前 | 変更後 | 差分 |
|------|--------|--------|------|
| Region coverage | 93.72% (23 missed / 366 regions) | 94.56% (27 missed / 496 regions) | **+0.84 pp** |
| Line coverage | 99.48% (1 missed / 193 lines) | 99.63% (1 missed / 271 lines) | +0.15 pp |
| Function coverage | 100% (0 missed / 18 fns) | 100% (0 missed / 26 fns) | — |

## 追加したテスト（8 本）

| テスト名 | 対象 |
|----------|------|
| `pdf_tag_to_krilla_tag_heading_level_zero_clamped_to_one` | level=0 → `clamp(1,6)` 下限分岐 |
| `pdf_tag_to_krilla_tag_heading_level_above_six_clamped` | level=7/10/255 → `clamp(1,6)` 上限分岐 |
| `pdf_tag_to_krilla_tag_heading_level_clamping_with_title` | クランプパスで `heading_title` を伴うケース |
| `semantic_entry_construction_and_clone` | `SemanticEntry` フィールド検証・`Clone` |
| `pdf_tag_partial_eq_and_clone` | `PdfTag` `PartialEq`/`Clone` derives。`ListNumbering` と `TableHeaderScope` の非等値ケースも網羅 |
| `pdf_tag_debug_is_non_empty` | `Debug` derive のスモークテスト |
| `classify_element_is_case_sensitive` | 大文字 HTML タグ名（`"P"`, `"H1"` など）が `None` を返すことを確認 |
| `classify_element_none_for_interactive_and_metadata_elements` | `form`, `input`, `video`, `iframe` など 22 要素が `None` を返すことを確認 |

## 意図的に除外したカバレッジ

- **`matches!` マクロの `_ => false` アーム**: 正しいテストでは常に到達不能（unreachable）。追加テストを増やすほど missed regions の絶対数も増えるが、分母も増えるため % は改善する。根本的な解消は不可能。
- **残り 1 missed line**: `pdf_tag_to_krilla_tag_th_scope_variants` 内の `if let TagKind::TH(tag) = k { ... }` の else ブランチ。正しいテストでは `TH` が必ず返るため構造上到達不能。既存テストの書き換えが必要であり、本 PR のスコープ外。
- **`extract_text_content` / `collect_text`**: `blitz_dom::BaseDocument` のインスタンス化が必要で Blitz 依存。本ルーティンはテスト専用のためプロダクションコード変更なし。

## チェックリスト

- [x] `cargo test -p fulgur --lib` — 1733 tests passed
- [x] `cargo clippy -p fulgur -- -D warnings` — 警告なし
- [x] `cargo fmt --check` — 差分なし
- [x] プロダクションコードの変更なし（テストのみ）

---
_Generated by [Claude Code](https://claude.ai/code/session_01VE54AsDtaqmhuMxvE5jo2H)_

## Summary by CodeRabbit

- **テスト**
  - 見出しレベル `H { level }` の変換が正しく検証され、`level=0` の補正や `6` 超のクランプ、さらにタイトル連携の経路も確認。
  - `SemanticEntry`／`PdfTag` の `clone`、等価性判定、デバッグ出力が適切であること、`alt_text = Some("")` が保持されることを検証。
  - 要素分類のエッジケースとして、ローカル名の大小文字一致と、インタラクティブ／非構造要素（フォーム等）で `None` を返すことを確認。

### CODE REVIEW

File: crates/fulgur/src/tagging.rs [hunk 1]
```diff
@@ -369,4 +369,155 @@ mod tests {
...
+            tag: PdfTag::Figure,
+            parent: None,
+            alt_text: Some(String::new()),
+        };
+        assert_eq!(e3.alt_text.as_deref(), Some(""));
+
+        // Clone reproduces all fields.
+        let e2c = e2.clone();
+        assert_eq!(e2c.parent, e2.parent);
+        assert_eq!(e2c.alt_text, e2.alt_text);
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/tagging.rs:447]: ご指摘の通りです。`tag` フィールドのアサーションが抜けていました。`assert_eq!(e2c.tag, e2.tag)` を追加して `c1f6255` でプッシュしました。

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/fulgur/src/tagging.rs [hunk 2]
```diff
@@ -369,4 +369,156 @@ mod tests {
...
+
+    #[test]
+    fn pdf_tag_to_krilla_tag_heading_level_clamping_with_title() {
+        // Clamping path with a heading_title to confirm both clamp and title flow.
+        let title = Some("Appendix".to_owned());
+        let k_low = pdf_tag_to_krilla_tag(&PdfTag::H { level: 0 }, title.clone(), None);
+        let k_high = pdf_tag_to_krilla_tag(&PdfTag::H { level: 9 }, title, None);
+        assert!(matches!(k_low, krilla::tagging::TagKind::Hn(_)));
+        assert!(matches!(k_high, krilla::tagging::TagKind::Hn(_)));
+    }
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/tagging.rs:411]: ご指摘の通りです。`Tag<kind::Hn>` に `level() -> NonZeroU16` が生えていることを krilla 0.7.0 の生成コード（`generated.rs:2098`）で確認しました。

`let krilla::tagging::TagKind::Hn(tag) = k else { panic!(...) }` パターンで分解し、`assert_eq!(tag.level().get(), expected)` で境界値を明示的に検証するよう 3 本のテストすべてを更新して `f7ccfc4` でプッシュしました。

---
_Generated by [Claude Code](https://claude.ai/code)_
